src/rules.ts

```diff
@@ -698,15 +718,17 @@ export const rules: Rule[] = [
         return ignoreCodeBlocksAndYAML(text, (text) => {
           const footnotes = text.match(/^\[\^\w+\]: .*$/gm); // collect footnotes
           if (footnotes != null) {
+            // ensures footnotes at the end of the document are recognized properly
+            if (text.slice(-1) != '\n') text += '\n';
+
             // remove footnotes that are their own paragraph
-            text = text.replace(/\n\n\[\^\w+\]: .*\n\n/gm, '\n\n');
+            text = text.replace(/\n(?:\n\[\^\w+\]: .*)+\n\n/gm, '\n\n');
 
             // remove footnotes directly before/after a line of text
-            text = text.replace(/\n?\n\[\^\w+\]: .*\n\n?/gm, '\n');
+            text = text.replace(/\n?(?:\n\[\^\w+\]: .*)+\n?/gm, '\n');
 
-            // remove footnotes sourrounded by text
-            text = text.replace(/\n\[\^\w+\]: .*\n/gm, '');
-            text += '\n\n' + footnotes.join('\n'); // append footnotes at the very end of the note
+            // append at the bottom
+            text = text.replace(/\n*$/, '') + '\n\n' + footnotes.join('\n'); // append footnotes at the very end of the note
```

Comment:
I assume these changes are to fix a bug? Can you add the previously failing test to tests.ts?

Reply:
yeah, they are bug fixes for #71.

Reply:
same as above, I lack knowledge about tests to do that, I am afraid